src/ObjectStore.jl

```diff
@@ -38,7 +38,10 @@ function init_rust_store(config::RustStoreConfig = RustStoreConfig(15, 150))
         if RUST_STORE_STARTED[]
             return
         end
-        @ccall rust_lib.start(config::RustStoreConfig)::Cint
+        res = @ccall rust_lib.start(config::RustStoreConfig)::Cint
+        if res != 0
+            error("Failed to init_rust_store")
```

Comment:
i wonder if we should throw a custom error here, to make downstream usage easier?

e.g. so users can do
```julia
try 
    init_rust_store(config)
catch err
    if err isa InitRustStoreException
       throw_helpful_error_with_advice()
    else  # something else went wrong!
      rethrow()
    end
end
```
Also maybe we should include that helpful advice here?
Can we say anything about why this might have failed? Is it just multiple init calls?